v2/pkg/engine/datasource/pubsub_datasource/pubsub_datasource.go

```diff
@@ -334,10 +297,117 @@ type RequestDataSource struct {
 }
 
 func (s *RequestDataSource) Load(ctx context.Context, input []byte, w io.Writer) error {
-	topic, err := jsonparser.GetString(input, "topic")
+	subject, err := jsonparser.GetString(input, "subject")
 	if err != nil {
 		return err
 	}
 
-	return s.pubSub.Request(ctx, topic, nil, w)
+	return s.pubSub.Request(ctx, subject, nil, w)
+}
+
+func (p *Planner[T]) extractEventSubject(ref int, subject string) (string, error) {
+	matches := eventSubjectRegex.FindAllStringSubmatch(subject, -1)
+	if len(matches) != 1 || len(matches[0]) != 2 {
+		return "", fmt.Errorf("expected subject to match regex")
+	}
+	argumentName := matches[0][1]
```

Comment:
Will this not panic when 310 is valid in case len(matches) = 1?
It always needs to be len = 2 to not panic here

Reply:
I don't understand, sorry.

Reply:
The first slice is at least size 1, so index 0 is accessed. The inner slice is at least slice 2, so index 1 is accessed.

Reply:
ok, found my mistake